block/manager.go

```diff
@@ -713,7 +733,7 @@ func (m *Manager) submitBlockToDA(ctx context.Context, block *types.Block) error
 	if !submitted {
 		return fmt.Errorf("failed to submit block to DA layer after %d attempts", maxSubmitAttempts)
 	}
-
+	m.pendingBlocks = make([]*types.Block, 0)
```

Comment:
I think we should move this to `NewManager`, in the odd case where `publishBlock` tries to push a block but panics because it's not initialized.

Reply:
OK, I see that we're re-initializing it here, so all good.